vital/types/detected_object.cxx

```diff
@@ -199,4 +199,58 @@ ::set_descriptor( descriptor_scptr d )
 }
 
 
+// ------------------------------------------------------------------
+std::vector< std::string >
+detected_object
+::notes() const
+{
+  return m_notes;
+}
+
+
+// ------------------------------------------------------------------
+void
+detected_object
+::add_note( const std::string& note )
+{
+  m_notes.push_back( note );
+}
+
+
+// ------------------------------------------------------------------
+void
+detected_object
+::reset_notes()
+{
+  m_notes.clear();
+}
+
+
+// ------------------------------------------------------------------
+std::map< std::string, vital::point_2d >
+detected_object
+::keypoints() const
+{
+  return m_keypoints;
+}
+
+
+// ------------------------------------------------------------------
+void
+detected_object
+::add_keypoint( const std::string id, vital::point_2d p )
```

Comment:
Did you mean to pass `id` by reference? What about `p`? (Also, [[types.qualified]](../tree/master/doc/code-style.rst).)

Reply:
p is a pretty simple type, but no preference I added one

Reply:
Yeah, could go either way with `p`. Best would be to get advice from clang-tidy, but don't worry about it. I was mainly suspicious of `id`.